lib/components/base/Button.vue

```diff
@@ -1,50 +1,45 @@
-<script setup>
+<script setup lang="ts">
 import { ExternalIcon, UnknownIcon } from '@'
 
 import { computed } from 'vue'
 
-const props = defineProps({
-  link: {
-    type: String,
-    default: null,
-  },
-  external: {
-    type: Boolean,
-    default: false,
-  },
-  action: {
-    type: Function,
-    default: null,
-  },
-  color: {
-    type: String,
-    default: 'default',
-  },
-  iconOnly: {
-    type: Boolean,
-    default: false,
-  },
-  large: {
-    type: Boolean,
-    default: false,
-  },
-  outline: {
-    type: Boolean,
-    default: false,
-  },
-  transparent: {
-    type: Boolean,
-    default: false,
-  },
-  hoverFilled: {
-    type: Boolean,
-    default: false,
-  },
-  hoverFilledOnly: {
-    type: Boolean,
-    default: false,
-  },
-})
+const props = withDefaults(
+  defineProps<{
+    link?: string
+    external: boolean
+    action?: (event: MouseEvent) => void
+    color:
+      | 'default'
+      | 'danger'
+      | 'primary'
+      | 'red'
+      | 'orange'
+      | 'green'
+      | 'blue'
+      | 'purple'
+      | 'gray'
+      | 'secondary'
+      | 'highlight'
+    iconOnly: boolean
+    large: boolean
+    outline: boolean
+    transparent: boolean
+    hoverFilled: boolean
+    hoverFilledOnly: boolean
+  }>(),
+  {
+    link: undefined,
```

Comment:
Wouldn't it be `undefined` anyway if you skipped it?

Reply:
did decide to keep it based on the discussion on discord